### PR TITLE
Added a script for protecting GitHub branches

### PR DESCRIPTION
**Pull request by lslezak**
This is just an one shot script (run&forget), I'm adding it here just for the future reference.

So don't be nitpicking in the review :wink:

### CODE REVIEW

File: github/protect_branches/protect_branches.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# --------------------------------------------------------------------
+
+# the GitHub organization, all organization repos will be processed
+# if you want to touch only some of them then write some filtering code
+GH_ORG = "yast"
+
+# install missing gems
+if !File.exist?("./.vendor")
+  puts "Installing the needed Rubygems to ./.vendor/bundle ..."
+  `bundle install --path .vendor/bundle`
```
Comment (jreidinger) [inline github/protect_branches/protect_branches.rb]: why not use system? so you see output?

File: github/protect_branches/protect_branches.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  # SLE11 GA
+  /\ACode-11\z/,
+  # SLE11 SPx
+  /\ACode-11-SP[0-9]+\z/,
+  # SLE12 GA
+  /\ASLE-12-GA\z/,
+  # SLE12 SPx
+  /\ASLE-12-SP[0-9]+\z/,
+  # CASP 1.0
+  /\ASLE-12-SP2-CASP\z/
```
Comment (jreidinger) [inline github/protect_branches/protect_branches.rb:75]: can we use this script in future for e.g. add SLE-13 branch?

Comment (lslezak) [reply github/protect_branches/protect_branches.rb:75]: Yes, just add a new branch into the list. (For a faster run you could remove or comment out the old branches to not set already protected branches.)

**Review by jreidinger — APPROVED** [review]
LGTM

**Pull request merged**